File: sagatave/nat_sk.py
def getNatural():
    while True:
        try:
            n = int(input("Ievadiet naturālu skaitli: "))
            if n > 0:
                break
            else:
                print("Ievadiet naturālu skaitli! (>0)\n")
        except ValueError:
            print("Ievadiet naturālu skaitli! (>0)\n")

    return n

while True:

    x = getNatural()

    turpinat = int(input("Velreiz? (1/0)"))
    if turpinat != 1:
        break

File: sagatave/nat_sk.cpp
#include <iostream>
using namespace std;

/*
int getNatural():
Atgriež kā rezultātu naturālu skaitli n.
*/

int getNatural() {
    int n;

    do {
        cout << "Ievadiet naturalu skaitli (>0): ";
            cin >> n;
            if (cin.fail() || n <= 0) {
                cout << "Nav naturals sk.(>0)" << endl << endl;
                cin.clear();
                cin.ignore(1000, '\n');
            } else break;

    } while (true);

    return n;
}

int main() {
    int ok;

    do {
        int atgriez = getNatural();
        cout << "Jūs ievadījāt: " << atgriez << endl;

        cout << "Vēlreiz? (1/0): ";
        cin >> ok;

    } while (ok == 1);
    return 0;
}


